File: pytools/examples/java/server/protocol/MsgSceneVector3.java
package protocol;


import network.Packet;
import network.PacketUtil;

import java.util.List;
import java.util.ArrayList;


public class MsgSceneVector3
{
	private short x;
	private short y;
	private short z;


	public MsgSceneVector3()
	{

	}

	public MsgSceneVector3(Packet packet)
	{
		x = packet.readShort();
		y = packet.readShort();
		z = packet.readShort();
	}

	public byte[] encode()
	{
		Packet packet = new Packet();
		packet.writeShort(x);
		packet.writeShort(y);
		packet.writeShort(z);
		return packet.readBytes();
	}

	public byte[] getBytes()
	{
		return this.encode();
	}



	public void setX(short x)
	{
		this.x = x;
	}
	public short getX()
	{
		return this.x;
	}

	public void setY(short y)
	{
		this.y = y;
	}
	public short getY()
	{
		return this.y;
	}

	public void setZ(short z)
	{
		this.z = z;
	}
	public short getZ()
	{
		return this.z;
	}

}

File: pytools/examples/php/server/protocol/MsgSceneVector3.php
<?php
namespace protocol;

use network\Packet;


class MsgSceneVector3
{
	private $x = 0;
	private $y = 0;
	private $z = 0;


	public function __construct(&$packet=null)
	{
		if ($packet)
		{
			$this->decode($packet);
		}
	}

	public function decode(&$packet)
	{
		$this->x = $packet->readI16();
		$this->y = $packet->readI16();
		$this->z = $packet->readI16();
	}

	public function encode()
	{
		$packet = new Packet();
		$packet->writeI16($this->x);
		$packet->writeI16($this->y);
		$packet->writeI16($this->z);
		return $packet->readBytes();
	}

	public function getBytes()
	{
		return $this->encode();
	}


	public function setX($x)
	{
		$this->x = $x;
	}
	public function getX()
	{
		return $this->x;
	}

	public function setY($y)
	{
		$this->y = $y;
	}
	public function getY()
	{
		return $this->y;
	}

	public function setZ($z)
	{
		$this->z = $z;
	}
	public function getZ()
	{
		return $this->z;
	}

}
